src/deye_daemon.py

```diff
@@ -78,25 +77,23 @@ def __init__(self, config: DeyeConfig):
                 "enabled" if set_time_processor.get_id() in config.active_processors else "disabled"
             )
         )
-        self.__last_observations: list[DeyeEvent] = []
+        self.__last_observations = DeyeEventList()
         self.__event_updated = time.time()
-        # ToDO: make the expiry time be configurable
-        self.__event_expiry = 360  # expiry time in seconds , default 6 min
 
     def do_task(self):
         self.__log.info("Reading start")
         regs = {}
         for reg_range in self.reg_ranges:
             self.__log.info(f"Reading registers [{reg_range}]")
             regs |= self.modbus.read_registers(reg_range.first_reg_address, reg_range.last_reg_address)
-        events: list[DeyeEvent] = []
+        events = DeyeEventList()
         events.append(DeyeLoggerStatusEvent(len(regs) > 0))
         events += self.__get_observations_from_reg_values(regs)
-        if self.__new_event(events):
+        if not self.__config.publish_on_change or self.__is_device_observation_changed(events):
             for processor in self.processors:
                 processor.process(events)
         else:
-            self.__log.info("Received event data is not new, skipping")
+            self.__log.info("Received event data is not new or logger is offline, skipping")
```

Comment:
A minor thing. I think the data is "new", but the measurements haven't changed, so I suggest to rephrase this log message as "No changes found in received data, or the logger is offline"

Reply:
Agree that this is more clear, will change it as suggested.